scandium-core/src/main/java/org/eclipse/californium/scandium/DtlsBindingPreprocessingConnector.java

```diff
@@ -0,0 +1,156 @@
+/*******************************************************************************
+ * Copyright (c) 2022 AVSystem and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v20.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *    Jakub Pinowski (AVSystem) - initial creation
+ ******************************************************************************/
+
+package org.eclipse.californium.scandium;
+
+import org.eclipse.californium.scandium.config.DtlsConnectorConfig;
+import org.eclipse.californium.scandium.dtls.ConnectionId;
+import org.eclipse.californium.scandium.dtls.Record;
+import org.eclipse.californium.scandium.dtls.ResumptionSupportingConnectionStore;
+
+import java.net.InetSocketAddress;
+import java.util.List;
+import java.util.concurrent.Executor;
+
+/**
+ * DTLS connector with customizable preprocessing of a connection binding.
+ * <p>
+ * Provides a way to perform external preprocessing using source socket address
+ * and connection ID before DTLSConnector's internal connection store
+ * is accessed.
+ * </p>
+ * <p>
+ * Example of usage; simple reading and restoring connection from shared storage.
+ * This code uses a separate ForkJoinPool to execute its action in order to release connector's
+ * receiver threads in case of some complex operations (reading from map is here to
+ * just show the concept).
+ * </p>
+ * <pre>
+ * final Set<ConnectionId> knownCids = new HashSet<>();
```

Comment:
I like it, but javadoc complains about `<` and `>` ;-(.
Please replace them by "&lt;" and "&lt;"

(And please add a hint, that `->` lamdas require to use at least java 8.)